packages/hop-node/src/db/TransferRootsDb.ts

```diff
@@ -172,6 +173,44 @@ class TransferRootsDb extends BaseDb {
       )
     })
   }
+
+  async getUnsettledTransferRoots (
+    filter: Partial<TransferRoot> = {}
+  ): Promise<TransferRoot[]> {
+    const transfers: TransferRoot[] = await this.getTransferRoots()
+    return transfers.filter(item => {
+      if (filter?.destinationChainId) {
+        if (filter.destinationChainId !== item.destinationChainId) {
+          return false
+        }
+      }
+
+      let rootSetTimestampOk = true
```

Comment:
I believe this timestamp was introduced to handle the case where a transfer from xDai -> L1 also commits the root hash. In this case, we were waiting x blocks for reorg, but the commit got sent to L1 before we were bonded the transfer. With our old software, we did not re-settle if a transfer got bonded after the first settlement.

We have updated the software in a few ways since this was introduced:
* We now resettled transferRoots if they should be resettled
* We now only wait 12 blocks on xDai instead of 24. This matches that of the Omni bridge and I believe fixes this issue

Given that, what are your thoughts on this piece of code? I don't _love_ it, but I think in a crazy edge case it could save us an L1 tx. On the downside, it makes settling and debugging slightly longer and more annoying. If we keep it, maybe we either cut the time in 1/2 or scope it to `if (item?.rootSetTimestamp && this.chainSlug === xDai)`. What are your thoughts on all that?

Reply:
I like the latter suggestion to scope to xdai with a shorter time to prevent making extra L1 txs